aws/asg.go

```diff
@@ -85,30 +89,86 @@ func getAutoScalingGroupsByName(service autoscalingiface.AutoScalingAPI, autoSca
 	return result, nil
 }
 
+// We look to the same string used for instance filtering, however, we are much more limited in what can be done for
+// ASGs. As such, we instead build a map of tags to look for as we iterate over all ASGs in getOwnedAutoScalingGroups
+func parseFilterTags(clusterId string) map[string][]string {
+	if filter, ok := os.LookupEnv(customTagFilterEnvVarName); ok {
```

Comment:
I would prefer if we could handle `os.LookupEnv` and value parsing in the beginning of the program i.e. close to `main` func where we also parse the flags. This allows us to keep such functions as these pure and simpler to test.

Given that we already have something similar here: https://github.com/zalando-incubator/kube-ingress-aws-controller/blob/bd660aa5cbeafd030572afc083c8aba21c73d81a/aws/adapter.go#L686 I'm fine not fixing it in this PR. but I would like to fix it afterwards at least.

Reply:
I could create a new field in the adapter manifest, and the parsing to adapter.go, and add a param to getOwnedAutoScalingGroups for the parsed version to work with?

Reply:
I would handle the parsing as early as here like we do for other configuration: https://github.com/zalando-incubator/kube-ingress-aws-controller/blob/bd660aa5cbeafd030572afc083c8aba21c73d81a/controller.go#L67 and then pass the resulting map to the adapter like you suggest.